src/stage.cpp

```diff
@@ -1,6 +1,31 @@
 #include "stage.h"
 
 //=============================================================================
+// Helpers
+//=============================================================================
+bool isObstacle(int prevTileValue, int tileValue) {
+	return
+		(prevTileValue == 1 && tileValue == 2) ||
+		(prevTileValue == 2 && tileValue == 1) ||
+		tileValue == 0 || tileValue == 3 || 
+		tileValue == 4 || tileValue == 5 || 
+		tileValue == 6;
+}
+
+bool isPath(int prevTileValue, int tileValue) {
+	bool valid = false;
+
+	if (prevTileValue == 1) 
+		valid = tileValue == 1 || tileValue == 3;
+	else if (prevTileValue == 2)
+		valid = tileValue == 2 || tileValue == 3;
+	else
+		valid = tileValue == 1 || tileValue == 2 || tileValue == 3;
+
+	return valid;
+}
```

Comment:
Not sure if `isObstacle` and `isPath` should be done on the specific entity classes instead. Since, different entities can move on different areas and every entity can access the stage class. Thoughts?